src/web/FrontLayerViewManager.tsx

```diff
@@ -88,6 +91,16 @@ export class FrontLayerViewManager {
             }
         }
 
+        if (options.cacheable) {
+            // New popup is transitioning from cached to active.
+            this._cachedPopups = this._cachedPopups.filter(popup => popup.id !== popupId);
+        }
+        if (this._activePopupOptions && this._activePopupOptions.cacheable) {
+            // Old popup is transitioning from active to cached.
+            this._cachedPopups.push({ options: this._activePopupOptions, id: this._activePopupId!!! });
+            this._cachedPopups = this._cachedPopups.slice(-MAX_CACHED_POPUPS);
```

Comment:
Are you sure slice() does the right thing if you pass a negative value that's greater than the current length of the array? I think it does, but the documentation for the method doesn't specify this case.

Reply:
ECMA 262 22.1.3.23 Array.prototype.slice ( start, end )
...
4. If relativeStart < 0, let k be max((len + relativeStart), 0); else let k be min(relativeStart, len).
...

Based on this I'd say that it is well specified. If this language spec is not relevant for this project I'll fix it.